go/runtime/txpool/txpool.go

```diff
@@ -302,33 +312,52 @@ func (t *txPool) PromoteProposedBatch(batch []hash.Hash) {
 	t.proposedTxsLock.Lock()
 	defer t.proposedTxsLock.Unlock()
 
-	for _, tx := range txs {
-		if tx == nil {
+	for i, h := range batch {
+		if _, ok := missingTxs[h]; ok {
 			continue
 		}
-		t.proposedTxs[tx.hash] = tx
+		t.proposedTxs[h] = txs[i]
 	}
 }
 
 func (t *txPool) ClearProposedBatch() {
 	t.proposedTxsLock.Lock()
 	defer t.proposedTxsLock.Unlock()
 
-	t.proposedTxs = make(map[hash.Hash]*Transaction)
+	t.proposedTxs = make(map[hash.Hash][]byte)
 }
 
-func (t *txPool) RemoveTxBatch(txs []hash.Hash) {
-	t.schedulerQueue.remove(txs)
-
-	pendingScheduleSize.With(t.getMetricLabels()).Set(float64(t.schedulerQueue.size()))
+func (t *txPool) GetSchedulingSuggestion() [][]byte {
+	var txs [][]byte
+	for _, q := range t.usableSources {
+		txs = append(txs, q.GetSchedulingSuggestion()...)
+	}
+	return txs
 }
 
-func (t *txPool) GetPrioritizedBatch(offset *hash.Hash, limit uint32) []*Transaction {
```

Comment:
Where did support for iterating over the prioritized suggested batch via offset/limit go? This is needed as we don't want to send the entire transaction queue into the runtime.

Reply:
gonna merge this with the discussion about UsableTransactionSource.GetSchedulingSuggestion